app/controllers/application_controller.rb

```diff
@@ -392,4 +400,14 @@ def mobile?
     client = DeviceDetector.new(request.user_agent)
     client.device_type != 'desktop'
   end
+
+  def update_session_paths_visited_for_analytics
+    session[:paths_visited] ||= {}
+    session[:first_path_visit] = !session[:paths_visited].key?(request.path)
+    session[:paths_visited][request.path] = true
+  end
+
+  def is_new_session_path?
```

Comment:
I meant rename this method too (and the analytics property), and I think it's conventional to avoid the `is_`:

```suggestion
  def first_path_visit?
```

and in retrospect, maybe something even more verbose?

```suggestion
  def first_path_visit_this_session?
```